cli/popper/commands/cmd_matrix.py

```diff
@@ -0,0 +1,58 @@
+import click
+import popper.utils as pu
+
+from popper.cli import pass_context
+from popper.exceptions import BadArgumentUsage
+
+
+@click.command('matrix', short_help='Define execution matrix for a '
+               'pipeline using environment variables.')
+@click.argument('pipeline', required=True)
+@click.option(
+    '--add',
+    multiple=True,
+    help='Add environment variable in the form key=value',
+    required=False
+)
+@click.option(
+    '--rm',
+    multiple=True,
+    help='Remove environment variable using the key',
+    required=False
+)
+@pass_context
+def cli(ctx, pipeline, add, rm):
+    try:
+        config = pu.read_config()
+        pipeline_config = config['pipelines'][pipeline]
+    except KeyError:
+        raise BadArgumentUsage("Pipeline {} does not exist.".format(pipeline))
```

Comment:
maybe we should add a `pu.read_pipeline_config` as a convenience for doing all this? There are other few places where we do this

Reply:
I need to read the whole config anyways since I have to write the changes back. So, there is no requirement of creating an extra utility function for this imo.